src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1047 @@
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      strncpy (ddl_audit_handle->sql_text, sql_text, len);
+      ddl_audit_handle->sql_text[len - 1] = '\0';
+    }
+}
+
+
+void
+cub_ddl_log_stmt_type (int stmt_type)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->stmt_type = stmt_type;
+}
+
+void
+cub_ddl_log_loaddb_file_type (char file_type)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->loaddb_file_type = file_type;
+}
+
+void
+cub_ddl_log_file_name (const char *file_name)
+{
+  if (ddl_audit_handle == NULL || file_name == NULL)
+    {
+      return;
+    }
+
+  strncpy (ddl_audit_handle->file_name, file_name, PATH_MAX);
+}
+
+void
+cub_ddl_log_file_line (int file_line)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->file_line_number = file_line;
+}
+
+void
+cub_ddl_log_err_msg (char *msg)
+{
+  if (ddl_audit_handle == NULL || msg == NULL)
+    {
+      return;
+    }
+
+  if (ddl_audit_handle->err_msg != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->err_msg);
+    }
+
+  ALLOC_COPY (ddl_audit_handle->err_msg, msg);
+  cub_remove_char (ddl_audit_handle->err_msg, '\n');
+}
+
+void
+cub_ddl_log_err_code (int err_code)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->err_code = err_code;
+}
+
+void
+cub_ddl_log_start_time (struct timeval *time_val)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->execute_start_time[0] = '\0';
+  cub_get_time_string (ddl_audit_handle->execute_start_time, time_val);
+}
+
+void
+cub_ddl_log_elapsed_time (long sec, long msec)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->elapsed_time, 20, "elapsed time %ld.%03ld", sec, msec);
+}
+
+void
+cub_ddl_log_msg (const char *fmt, ...)
+{
+  va_list args;
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  va_start (args, fmt);
+  vsnprintf (ddl_audit_handle->msg, DDL_LOG_MSG, fmt, args);
+  va_end (args);
+}
+
+void
+cub_ddl_log_type (char type)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->log_type = type;
+}
+
+void
+cub_ddl_log_execute_result (T_SRV_HANDLE * srv_handle)
+{
+  if (srv_handle == NULL)
+    {
+      return;
+    }
+
+  for (int i = 0; i < srv_handle->num_q_result; i++)
+    {
+      if (cub_is_ddl_type (srv_handle->q_result[i].stmt_type) == TRUE)
+	{
+	  cub_ddl_log_stmt_type (srv_handle->q_result[i].stmt_type);
+	  return;
+	}
+      else
+	{
+	  cub_ddl_log_stmt_type (srv_handle->q_result[i].stmt_type);
+	}
+    }
+}
+
+static int
+cub_file_copy (char *src_file, char *dest_file)
+{
+  char buf[FILE_BUFFER_SIZE] = { 0 };
+  size_t size;
+  int retval = 0;
+
+  if (src_file == NULL || dest_file == NULL)
+    {
+      return 0;
+    }
+
+  FILE *fsource = fopen (src_file, "r");
+  FILE *fdest = fopen (dest_file, "w");
```

Comment:
Please check whether **fsource** or **fdest** is NULL.

Reply:
Thank you, i will change code as your recommend.